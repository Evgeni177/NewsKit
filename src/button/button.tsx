import React from 'react';
import {ButtonProps, ButtonSize} from './types';
import {StyledFlag} from './styled';
import {useTheme} from '../theme';
import {filterOutFalsyProperties} from '../utils/filter-object';
import {as as emotionAs} from '../utils/component';
import {IndeterminateProgressIndicator} from '../icons/filled/custom/indeterminate-progress-indicator';
import {useInstrumentation, EventTrigger} from '../instrumentation';

export const Button: React.FC<ButtonProps> = ({
  children,
  overrides = {},
  ...props
}) => {
  const theme = useTheme();
  const {fireEvent} = useInstrumentation();
  const {size = ButtonSize.Medium} = props;
  const {disabled, loading, eventContext, eventOriginator = 'button'} = props;

  const buttonSettings: typeof overrides = {
    ...theme.componentDefaults.button[size],
    ...filterOutFalsyProperties(overrides),
  };

  return (
    <StyledFlag
      type="button"
      data-testid="button"
      disabled={disabled}
      loading={loading}
      {...emotionAs('button')}
      {...props}
      onClick={(...args) => {
        fireEvent({
          originator: eventOriginator,
          trigger: EventTrigger.Click,
          context: {
            ...eventContext,
          },
        });
        if (props.onClick) {
          props.onClick(...args);
        }
      }}
      overrides={buttonSettings}
    >
      {loading ? (
        <IndeterminateProgressIndicator
          overrides={{
            size: buttonSettings!.iconSize,
            stylePreset: buttonSettings.stylePreset as string,
          }}
        />
      ) : (
        children
      )}
    </StyledFlag>
  );
};
