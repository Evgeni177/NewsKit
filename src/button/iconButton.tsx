import React from 'react';
import {ButtonProps} from './types';
import {BaseButton} from './baseButton';

export const IconButton: React.FC<
  React.ButtonHTMLAttributes<HTMLButtonElement> & ButtonProps
> = ({children, equalSides, ...restOfProps}) => (
  <BaseButton equalSides {...restOfProps}>
    {children}
  </BaseButton>
);
