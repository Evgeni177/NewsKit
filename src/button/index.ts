export * from './button';
export * from './icon-button';
export {ButtonProps, ButtonSizing, ButtonSize} from './types';
