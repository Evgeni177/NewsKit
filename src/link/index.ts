export * from './types';
export * from './link';
export * from './link-standalone';
