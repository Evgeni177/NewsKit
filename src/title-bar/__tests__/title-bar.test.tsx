import React from 'react';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {TitleBar, TitleBarProps} from '..';
import {Button} from '../../button';
import {Link} from '../../link';

describe('TitleBar', () => {
  const TITLE = 'Test title of title bar';
  test('should display correctly with default styles and given title', () => {
    const fragment = renderToFragmentWithTheme(TitleBar, {
      children: TITLE,
    } as TitleBarProps);
    expect(fragment).toMatchSnapshot();
  });

  test('should apply correctly the custom presets ', () => {
    const fragment = renderToFragmentWithTheme(TitleBar, {
      children: TITLE,
      overrides: {
        stylePreset: 'standfirst',
        spaceInset: 'spaceInsetSquish010',
        heading: {
          typographyPreset: {
            xs: 'heading010',
            sm: 'heading020',
            md: 'heading030',
            lg: 'heading040',
          },
        },
      },
    } as TitleBarProps);
    expect(fragment).toMatchSnapshot();
  });

  describe('should display correctly as', () => {
    const headingLevels = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];

    test.each(headingLevels)('renders with %s size', currentSize => {
      const fragment = renderToFragmentWithTheme(TitleBar, {
        children: TITLE,
        headingAs: currentSize,
      } as TitleBarProps);
      expect(fragment).toMatchSnapshot();
    });
  });

  describe('Should display correctly with action item as:', () => {
    test('Button', () => {
      const button = () => <Button>Default button</Button>;
      const fragment = renderToFragmentWithTheme(TitleBar, {
        children: TITLE,
        actionItem: button,
      } as TitleBarProps);
      expect(fragment).toMatchSnapshot();
    });
    test('Link', () => {
      const link = () => <Link href="/">Link</Link>;
      const fragment = renderToFragmentWithTheme(TitleBar, {
        children: TITLE,
        actionItem: link,
      } as TitleBarProps);
      expect(fragment).toMatchSnapshot();
    });
  });
});
