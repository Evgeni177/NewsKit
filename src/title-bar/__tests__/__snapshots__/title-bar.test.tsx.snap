// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar Should display correctly with action item as: Button 1`] = `
<DocumentFragment>
  .emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-5 {
  padding-left: 24px;
  display: none;
}

@media screen and (min-width:480px) {
  .emotion-5 {
    display: block;
  }
}

.emotion-4 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  box-sizing: border-box;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
  margin: 8px 0;
}

.emotion-4:disabled {
  cursor: not-allowed;
}

.emotion-4 svg {
  fill: #FFFFFF;
  color: #FFFFFF;
}

.emotion-4:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-4:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-4:disabled {
  background-color: #f0f0f0;
  color: #9c9c9c;
}

.emotion-4:disabled svg {
  fill: #9c9c9c;
  color: #9c9c9c;
}

.emotion-4 svg {
  width: 16px;
  height: 16px;
}

.emotion-4::before {
  content: '';
  position: absolute;
  left: 0;
  top: -8px;
  width: 100%;
  height: 48px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
    <div
      class="emotion-5"
    >
      <button
        class="emotion-4"
        type="button"
      >
        <div
          class="emotion-3"
        >
          <div
            class="emotion-2"
          >
            Default button
          </div>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar Should display correctly with action item as: Link 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-3 {
  padding-left: 24px;
  display: none;
}

@media screen and (min-width:480px) {
  .emotion-3 {
    display: block;
  }
}

.emotion-2 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  color: #2F75D7;
}

.emotion-2:hover:not(:disabled) {
  color: #2F75D7;
}

.emotion-2:active:not(:disabled) {
  color: #2F75D7;
}

.emotion-2:visited:not(:disabled) {
  color: #903D91;
}

<div
    class="emotion-4"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
    <div
      class="emotion-3"
    >
      <a
        class="emotion-2"
        href="/"
      >
        Link
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading1 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 40px;
  line-height: 1.1;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 24px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h1
        class="emotion-0"
      >
        Test title of title bar
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading2 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading3 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 12px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h3
        class="emotion-0"
      >
        Test title of title bar
      </h3>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading4 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 22px;
  line-height: 1.0909090909090908;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 8px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h4
        class="emotion-0"
      >
        Test title of title bar
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading5 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 18px;
  line-height: 1.1111111111111112;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-size: 20px;
  line-height: 1.2;
  color: #0a0a0a;
  margin: 0 0 8px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h5
        class="emotion-0"
      >
        Test title of title bar
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly as Heading6 tag 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #0a0a0a;
  margin: 0 0 8px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h6
        class="emotion-0"
      >
        Test title of title bar
      </h6>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with align center 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with align left 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with border Bottom 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: solid;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with border None 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with border Top 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: solid;
  border-bottom-style: none;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with border TopAndBottom 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: solid;
  border-bottom-style: solid;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with default styles and given title 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with padding left 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TitleBar should display correctly with padding right 1`] = `
<DocumentFragment>
  .emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  color: #2e2e2e;
}

.emotion-0 {
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0a0a0a;
  margin: 0 0 16px 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-right: 24px;
  padding-bottom: 24px;
  box-sizing: border-box;
  border-color: #f0f0f0;
  border-width: 1px;
  border-style: none;
  border-top-style: none;
  border-bottom-style: none;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <h2
        class="emotion-0"
      >
        Test title of title bar
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
